### PR TITLE
show-dependencies can be expensive due to the resolve call

### PR DESCRIPTION
**Pull request by elsh**
Add an option to opt out of resolving if needed
rdar://100831624

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

File: Sources/Commands/SwiftPackageTool.swift [hunk 1]
```diff
@@ -749,8 +749,12 @@ extension SwiftPackageTool {
                 help: "The absolute or relative path to output the resolved dependency graph.")
         var outputPath: AbsolutePath?
 
+        @Flag(name: .long,
+              help: "Fetches or updates manifests to get the dependencies if not cached (can be expensive).")
+        var update: Bool = false
```
Comment (neonichu) [inline Sources/Commands/SwiftPackageTool.swift:754]: I feel like the default should continue to be true?

File: Sources/Workspace/Workspace.swift [hunk 2]
```diff
@@ -2435,6 +2437,7 @@ extension Workspace {
         root: PackageGraphRootInput,
         explicitProduct: String? = nil,
         forceResolution: Bool,
+        skipResolve: Bool = false,
```
Comment (tomerd) [inline Sources/Workspace/Workspace.swift:2440]: what is the difference between `skipResolve` and `forceResolution`, they seems to be related and the interaction between them is not clear to me

File: Sources/Commands/SwiftTool.swift [hunk 3]
```diff
@@ -618,13 +620,14 @@ public class SwiftTool {
...
                 forceResolvedVersions: options.resolver.forceResolvedVersions,
+                skipResolve: skipResolve,
                 testEntryPointPath: testEntryPointPath,
                 observabilityScope: self.observabilityScope
             )
 
             // Throw if there were errors when loading the graph.
             // The actual errors will be printed before exiting.
-            guard !self.observabilityScope.errorsReported else {
+            if self.observabilityScope.errorsReported && exitOnError {
```
Comment (tomerd) [inline Sources/Commands/SwiftTool.swift:630]: why is this change required?

File: Sources/Commands/SwiftPackageTool.swift (hunk 1 above)
Comment (tomerd) [reply Sources/Commands/SwiftPackageTool.swift:754]: I am also a bit confused about what this flag is supposed to determine. iiuc, the desired behavior of `show-dependencies` is to resolve dependences when they have not been resolved yet, but to skip it when they are already known/cached, which is just like `swift build` works. is this is how it works now? or does it always  resolve dependences, even if they are already supposed to be known? if so, this may be a bug, and not require a new flag

File: Sources/Commands/SwiftTool.swift (hunk 3 above)
Comment (elsh) [reply Sources/Commands/SwiftTool.swift:630]: In case package resolve is skipped with the new option, it can result in not finding a dependent product, which will then throw an error and then exits. This will ensure an error msg in dumpDependencies func is displayed.

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
Closing this based on comments above.
